up-l1/README.adoc

```diff
@@ -237,6 +237,7 @@ UTransport implementations
 * *MUST* support registering more than one listener for any given address patterns
 * *MUST* support registering the same listener for multiple address patterns
 * *MUST* document the maximum supported number of listeners per address pattern.
+* *MUST* be idempotent, multiple calls to said API with the same parameters *MUST* have the same effect as a single call.
```

Comment:
Practically speaking, does this mean just that if there were some sort of error other than ALREADY_EXISTS, we'd return that error and if we would have returned ALREADY_EXISTS, return OK?

Reply:
If we want all calls to be idempotent then IMHO `unregister` should never return `NOT_FOUND`, or should it?

Reply:
That makes sense to me @sophokles73. If it succeeds or is not found we still just return `OK`.

Reply:
I think there is value in the unregisterListener returning NOT_FOUND if you try and unregister something that is not there (that is an error) but registering the same listener twice just has the same affect as doing it once.